src/sagemaker_algorithm_toolkit/hyperparameter_validation.py

```diff
@@ -244,15 +244,26 @@ def _visit(name, visited, stack):
         return dependencies_stack
 
     def validate(self, user_hyperparameters):
-        # NOTE: 0. Validate required or fill in default.
+        # NOTE: 0. Convert possible aliases to original terms
+        tmp_user_hyperparameters = {}
+        hp_aliases = {"learning_rate": "eta", "min_split_loss": "gamma", "reg_lambda": "lambda", "reg_alpha": "alpha"}
+
+        for hp, value in user_hyperparameters.items():
+            if hp in hp_aliases:
+                tmp_user_hyperparameters[hp_aliases[hp]] = value
+            else:
+                tmp_user_hyperparameters[hp] = value
+        user_hyperparameters = tmp_user_hyperparameters
+
+        # NOTE: 1. Validate required or fill in default.
```

Comment:
This approach seems to handle aliases correctly. As an alternative, how about creating a method such as `_declare_alias(key_name, alias_name)` within the `Hyperparameters` class and use that to declare aliases when initializing hyperparametes? It may help with readability since aliases will be declared closer to where hps are initialized (as opposed to maintaining a list within the `validate` method itself). Similarly, moving logic to `_declare_alias()` will help with keeping code modular.

What do you think?

Reply:
I'm not sure how we can use `_declare_alias()` here, but I could I add an attribute `alias` to the `Hyperparameter` class. Through this we can declare aliases for our hyperparameters when we are initializing in https://github.com/aws/sagemaker-xgboost-container/blob/master/src/sagemaker_xgboost_container/algorithm_mode/hyperparameter_validation.py.

Reply:
my bad, I meant `declare_alias(key_name, alias_name)`, which would eventually add an alias to the `alias` attribute from the file you pointed out.